packages/website/docs/plugins.md

```diff
@@ -3,21 +3,244 @@ id: plugins
 title: Plugins
 ---
 
-:::note Draft
+Plugins encapsulate and distribute custom Autocomplete behaviors.
 
-This page needs to cover:
+An autocomplete can be much more than a functional combo box. **Autocomplete lets you extend and encapsulate custom behavior with its Plugin API.**
 
-- Plugins are a way of encapsulating sources, including the data and display. They can also trigger particular actions on an interaction.
-- For example, the insights plugin sends click and conversion events.
-- We provide some out-of-the-box plugins, but you can also build your own.
-- As a simple example, you could create a plugin to display a static list of predefined items:
-  - Code snippet
-- We also have created a couple out-of-the box plugins:
-  - Our **recent-searches plug-in** stores recently made searches in local storage so that you can incorporate an individual’s recent searches into an autocomplete.
-    - Code snippet + live example
-  - Our **query-suggestions plug-in** is made to work with an Algolia Query Suggestions index.
-    - Code snippet + live example
+For example, the [official Algolia Insights plugin](createAlgoliaInsightsPlugin) automatically sends click and conversion events to the [Algolia Insights API](https://www.algolia.com/doc/rest-api/insights/) whenever a user interacts with the autocomplete.
 
-We'll also need to document the Plugin API.
+You can use one of our existing official plugins or build your own.
+
+## Usage
+
+### Using an Autocomplete plugin
+
+When using an Autocomplete plugin, all you need to do is provide it via the `plugins` option.
+
+For example, when using the Insights plugin, you can instantiate the plugin, then pass it down to your Autocomplete instance.
+
+```js {11,15}
+import algoliasearch from 'algoliasearch/lite';
+import { autocomplete } from '@algolia/autocomplete-js';
+import { createAlgoliaInsightsPlugin } from '@algolia/autocomplete-plugin-algolia-insights';
+import insightsClient from 'search-insights';
+
+const appId = 'latency';
+const apiKey = '6be0576ff61c053d5f9a3225e2a90f76';
+const searchClient = algoliasearch(appId, apiKey);
+insightsClient('init', { appId, apiKey });
+
+const algoliaInsightsPlugin = createAlgoliaInsightsPlugin({ insightsClient });
+
+autocomplete({
+  // ...
+  plugins: [algoliaInsightsPlugin],
+});
+```
+:::note
+
+Plugins execute sequentially, in the defined order.
 
 :::
+
+### Building your own plugin
+
+An Autocomplete plugin is a simple object that implements the `AutocompletePlugin` interface.
+
+It can provide sources, react to state changes, and hook into various autocomplete lifecycle steps. It has access to setters, including the [Context API](context), allowing it to store and retrieve arbitrary data at any time.
```

Comment:
```suggestion
It can [provide sources](sources), [react to state changes](state), and hook into various autocomplete lifecycle steps. It has access to setters, including the [Context API](context), allowing it to store and retrieve arbitrary data at any time.
```